gtsfm/data_association/data_assoc.py

```diff
@@ -48,6 +51,16 @@ class DataAssociation(NamedTuple):
     triangulation_options: TriangulationOptions
     save_track_patches_viz: Optional[bool] = False
 
+    def get_ui_metadata() -> UiMetadata:
+        """Returns data needed to display this process in the process graph."""
+
+        return UiMetadata(
+            "Data Association",
+            "Multi-View Optimizer",
+            ("View-Graph Correspondences", "Global Rotations", "Global Translations", "Camera Intrinsics"),
```

Comment:
optional: would it be possible to discover these automatically from the function definition for `create_computaton_graph()`? that way they wouldn't go stale if we change that function def?

Reply:
This was my suggestion earlier too. The problem is that create_computation_graph does not have the type/name of returned arguments. run() would have types, but still not names. Without names, we won't know which output argument is fed as input to another node in the graph. Even IF we had argument names in these methods, we would have to keep them consistent across classes to infer the graph from them. 

Keeping this design, the only way would be make arguments and results derive a base class (for a grey node) which has some name metadata. I don't know if this is overkill.. depends on how we end up using this graph.

Reply:
@dellaert mentioned last Weds that he thought this was overkill, given the large infrastructure changes necessary that @akshay-krishnan has described.